optuna/integration/sklearn.py

```diff
@@ -0,0 +1,923 @@
+        """
+
+        self._check_is_fitted()
+
+        return self.best_estimator_.inverse_transform
+
+    @property
+    def predict(self):
+        # type: () -> Callable[..., Union[OneDimArrayType, TwoDimArrayType]]
+        """Call ``predict`` on the best estimator.
+
+        This is available only if the underlying estimator supports ``predict``
+        and ``refit`` is set to :obj:`True`.
+        """
+
+        self._check_is_fitted()
+
+        return self.best_estimator_.predict
+
+    @property
+    def predict_log_proba(self):
+        # type: () -> Callable[..., TwoDimArrayType]
+        """Call ``predict_log_proba`` on the best estimator.
+
+        This is available only if the underlying estimator supports
+        ``predict_log_proba`` and ``refit`` is set to :obj:`True`.
+        """
+
+        self._check_is_fitted()
+
+        return self.best_estimator_.predict_log_proba
+
+    @property
+    def predict_proba(self):
+        # type: () -> Callable[..., TwoDimArrayType]
+        """Call ``predict_proba`` on the best estimator.
+
+        This is available only if the underlying estimator supports
+        ``predict_proba`` and ``refit`` is set to :obj:`True`.
+        """
+
+        self._check_is_fitted()
+
+        return self.best_estimator_.predict_proba
+
+    @property
+    def score_samples(self):
+        # type: () -> Callable[..., OneDimArrayType]
+        """Call ``score_samples`` on the best estimator.
+
+        This is available only if the underlying estimator supports
+        ``score_samples`` and ``refit`` is set to :obj:`True`.
+        """
+
+        self._check_is_fitted()
+
+        return self.best_estimator_.score_samples
+
+    @property
+    def set_user_attr(self):
+        # type: () -> Callable[..., None]
+        """Call ``set_user_attr`` on the :class:`~optuna.study.Study`."""
+
+        self._check_is_fitted()
+
+        return self.study_.set_user_attr
+
+    @property
+    def transform(self):
+        # type: () -> Callable[..., TwoDimArrayType]
+        """Call ``transform`` on the best estimator.
+
+        This is available only if the underlying estimator supports
+        ``transform`` and ``refit`` is set to :obj:`True`.
+        """
+
+        self._check_is_fitted()
+
+        return self.best_estimator_.transform
+
+    @property
+    def trials_dataframe(self):
+        # type: () -> Callable[..., pd.DataFrame]
+        """Call ``trials_dataframe`` on the :class:`~optuna.study.Study`."""
+
+        self._check_is_fitted()
+
+        return self.study_.trials_dataframe
+
+    def __init__(
+        self,
+        estimator,  # type: BaseEstimator
+        param_distributions,  # type: Mapping[str, distributions.BaseDistribution]
+        cv=5,  # type: Union[int, BaseCrossValidator, None]
+        enable_pruning=False,  # type: bool
+        error_score=np.nan,  # type: Union[str, float]
+        load_if_exists=False,  # type: bool
```

Comment:
In my feeling, this constructor takes too many arguments.
How about passing `Study` instance directly and removing related arguments from the constructor?
(i.e., removes `load_if_exists`, `pruner`, `sampler`, `storage`, `study_name`, and adds `study` with the default value `None`)

I think it will improve maintainability too.  By passing `study` instance, we will not have to modify `OptunaSearchCV` when we add more arguments (or methods) to `Study` in the future.

Reply:
I removed those arguements.

Reply:
Thank you!